src/pandablocks_ioc/_hdf_ioc.py

```diff
@@ -106,6 +107,22 @@ def __init__(self, client: AsyncioClient, record_prefix: str):
             record_prefix + ":" + num_capture_record_name.upper()
         )
 
+        flush_mode_record_name = EpicsName(self._HDF5_PREFIX + ":FlushMode")
+        self._flush_mode_record = builder.mbbOut(
+            flush_mode_record_name,
+            *[flush_mode.name for flush_mode in FlushMode],
+            initial_value=0,
+        )
+        add_pvi_info(
+            PviGroup.INPUTS,
+            self._flush_mode_record,
+            flush_mode_record_name,
+            builder.mbbOut,
+        )
+        self._flush_mode_record.add_alias(
+            record_prefix + ":" + flush_mode_record_name.upper()
+        )
+
         flush_period_record_name = EpicsName(self._HDF5_PREFIX + ":FlushPeriod")
```

Comment:
How does this FlushPeriod look/behave now that we have the FlushMode? From a user's point of view, are they still going to see a non-zero FlushPeriod value, regardless of the value of the Mode? Is there some way we can reflect what records are valid configuration based on the selected Mode?

Reply:
I looked into these but they don't work for what we want:

![image](https://github.com/PandABlocks/PandABlocks-ioc/assets/21142790/8c5041f8-815d-4f75-adf8-14cb21fd28fc)

We would want something purely visual since the user should still be able to input into `FlushPeriod` before an upcoming change of mode. The best I can think of is some bool records that would appear next to the PVs to display that the input works for the current mode. This would require some PVI tweaking. I'm don't think this is a good idea though.

Reply:
Can we do anything with alarm states? e.g. when FlushMode is set to certain values, put some records into error states? May not be entirely intuitive though...

Pinging @coretl in case he has other ideas.

Reply:
I think using alarm states (MINOR or MAJOR I don't think we can add additional ones?) would be a mistake as we'd be using that feature in an exceptional way. But it would be handy if there was some feature on the EPICS level to handle this.

I certainly don't think it's unintuitive - the user will see that it's manual/periodic so will look for a button/period input.

Reply:
I think it is probably clear enough as it is:
- In immediate mode then it flushes every time, so period and flush now have no effect
- in manual mode then period has no effect, and flush only happens on flush now
- In periodic mode the period is honoured

The only question I have is what should happen if you click flush now in periodic mode. If we can make it actually do the flush then any combination of variables makes sense and we don't need to find a way to raise an error to the user.

Reply:
We could add another state to the FlushMode enum of "Hybrid" or somesuch. Although, if I understand correctly, it may be better to name it more specifically "PeriodicAndImmediate" or similar, as that's the two modes that we'd be combining.

I also realise that, if we go down that route, we're sort of invalidating the whole point of the enum in the first place as we would then want multiple different operation modes at the same time.

The other thought I had was to create another longStringOut PV, which is simply used to report back misconfigurations. If I understand correctly, there's no UI feedback if for example you set FlushMode to Immediate but also set a FlushPeriod.

Reply:
> The other thought I had was to create another longStringOut PV, which is simply used to report back misconfigurations. If I understand correctly, there's no UI feedback if for example you set FlushMode to Immediate but also set a FlushPeriod.

If we did this is I wouldn't want it to be for misconfigurations, it would be to display active PVs in the current mode. A user should be able to set `:FlushPeriod` while `IMMEDIATE` is activate, then hit stop capture and change the mode. I'm undecided on if such a display would be a good idea though. On the one hand you would be explicitly telling the user "you're in `PERIODIC`, you use the `:FlushPeriod` to control that", on the other hand I think it's already understandable enough and we may be adding unnecessary widgets.

> The only question I have is what should happen if you click flush now in periodic mode. If we can make it actually do the flush then any combination of variables makes sense and we don't need to find a way to raise an error to the user.

This was my conception of the button originally. It would be nice to offer the user this option.

> We could add another state to the FlushMode enum of "Hybrid" or somesuch. Although, if I understand correctly, it may be better to name it more specifically "PeriodicAndImmediate" 

It would be `PeriodicAndManual`. I think this change would complicate things more while trying to make them more understandable. I don't really see a reason why the FlushButton shouldn't also work in periodic? To make it clearer we could change `Manual` mode to be `OnlyManual`.